sstables/sstables.cc

```diff
@@ -2125,6 +2125,10 @@ sstable::make_reader(
     if (_version >= version_types::mc) {
         return mx::make_reader(shared_from_this(), std::move(schema), std::move(permit), range, slice, pc, std::move(trace_state), fwd, fwd_mr, mon);
     }
+
+    if (slice.options.contains(query::partition_slice::option::reversed)) {
+        on_internal_error(sstlog, "sstable::make_reader (v2 version): reversed slices not supported for sstable versions < mc");
+    }
```

Comment:
I "ensured" it by writing a comment on `sstable::make_reader` declaration:
```
    // Precondition: if sstable version is < mc, the slice is not reversed.
```

I can insert a reversing wrapper. Didn't because `make_reader` is only used in tests currently (production code uses `make_reader_v1`) and the codepath is dead but I guess it's better to have it conform to the general interface.

I'm also wondering what to do with reversed multi-partition queries for >= mc versions. I can also insert a reversing wrapper, but that will require something like:
```
upgrade_to_v2(make_reversing_reader(downgrade_to_v1(mx::make_reader(...)...)
```
ugly but I can do it, it doesn't really matter anyway since reversed multi-partition queries (available from Thrift only) are effectively dead.

Reply:
We're transitioning to v2 so soon the production code will be switched to use `make_reader()`. Better fix it now since the logic is hot in your cache.

Regarding multi-partition reversing, for now inserting converters is fine. Eventually make_reversing_reader() should be v2 natively and the converters will go away.